skrub/tests/test_deduplicate.py

```diff
@@ -95,3 +98,67 @@ def test__create_spelling_correction(seed: int = 123) -> None:
             spelling_correction.values[clusters == n].astype("int")
             == counts[clusters == n].max()
         ).all()
+
+
+@skip_if_no_parallel
+def test_parallelism():
+    # Test that parallelism works
+    X = make_deduplication_data(examples=['black', 'white', 'red'], entries_per_example=[500, 500, 500], prob_mistake_per_letter=0.3)
+    y = deduplicate(X, n_jobs=None)
+    times = {}
+    jobs = [1, 4]
+
+    for n_jobs in jobs:
+        start = time.time()
+        y_parallel = deduplicate(X, n_jobs=n_jobs)
+        assert_array_equal(y, y_parallel)
+        end = time.time()
+        timer = end - start
+        times[n_jobs] = timer
+    print(times)
+    # Test if execution time is shorter:
+    for i in range(len(jobs)-1):
+        assert times[jobs[i+1]] < times[jobs[i]]
+
+    # Test with threading backend
+    with joblib.parallel_backend("threading"):
+        y_threading = deduplicate(X, n_jobs=1)
+    assert_array_equal(y, y_threading)
+
+
+DEFAULT_JOBLIB_BACKEND = joblib.parallel.get_active_backend()[0].__class__
+
+
+class DummyBackend(DEFAULT_JOBLIB_BACKEND):  # type: ignore
+    """
+    A dummy backend used to check that specifying a backend works
+    in deduplicate.
+    The `count` attribute is used to check that the backend is used.
+    Copied from https://github.com/scikit-learn/scikit-learn/blob/36958fb240fbe435673a9e3c52e769f01f36bec0/sklearn/ensemble/tests/test_forest.py  # noqa
+    """
+    def __init__(self, *args, **kwargs):
+        self.count = 0
+        super().__init__(*args, **kwargs)
+
+    def start_call(self):
+        self.count += 1
+        return super().start_call()
+
+
+joblib.register_parallel_backend("testing", DummyBackend)
+
+
+@skip_if_no_parallel
+def test_backend_respected():
+    """
+    Test that the joblib backend is used.
+    Copied from https://github.com/scikit-learn/scikit-learn/blob/36958fb240fbe435673a9e3c52e769f01f36bec0/sklearn/ensemble/tests/test_forest.py  # noqa
+    """
+    # Test that parallelism works
+    X = make_deduplication_data(examples=['black', 'white'],
+                                entries_per_example=[15, 15])
+    deduplicate(X, n_jobs=2)
+
+    with joblib.parallel_backend("testing") as (ba, n_jobs):
```

Comment:
With the recent version of `joblib`, the use `parallel_config` is advised. (possible if you support `python3.8+`).